src/plugins/intel_cpu/src/nodes/mkldnn_deconv_node.cpp

```diff
@@ -592,35 +609,123 @@ void MKLDNNDeconvolutionNode::prepareParams() {
+                std::swap(weightDims[withGroups + 0], weightDims[withGroups + 1]);
+            }
+
+            auto inDesc = mkldnn::memory::desc(MKLDNNExtensionUtils::convertToDnnlDims(key.inp0->getShape().getStaticDims()),
+                                                                                       memory::data_type::f32,
+                                                                                       memory::format_tag::any);
+            auto wghDesc = mkldnn::memory::desc(weightDims,
+                                                memory::data_type::f32,
+                                                memory::format_tag::any);
+            auto outDesc = mkldnn::memory::desc(MKLDNNExtensionUtils::convertToDnnlDims(key.out->getShape().getStaticDims()),
+                                                                                        memory::data_type::f32,
+                                                                                        memory::format_tag::any);
+
+            std::shared_ptr<MKLDNNDescriptor> anyDeconvDesc = createDefaultMkldnnDeconvDesc(inDesc, wghDesc, outDesc, false,
+                                                                                key.stride, key.dilation, key.paddingL, key.paddingR, engine);
+            auto anyDeconvItpd = anyDeconvDesc->createPrimitiveDescriptorIterator(engine, key.attr);
+            if (static_cast<bool>(anyDeconvItpd)) {
+                auto prim_desc = convolution_backward_data::primitive_desc(anyDeconvItpd.get());
+                execPtr = std::make_shared<DeconvExecutorDefault>(prim_desc,
+                                                                  key.inp0->getDnnlDesc(),
+                                                                  (key.isInt8 ?
+                                                                  mkldnn::memory::desc(weightDims,
+                                                                                       memory::data_type::s8,
+                                                                                       MKLDNNExtensionUtils::GetPlainFormatByRank(weightDims.size())) :
+                                                                  key.inp1),
+                                                                  key.out->getDnnlDesc(),
+                                                                  engine);
+            }
+        }
+
+        return execPtr;
+    };
+
+    execPtr = nullptr;
+    auto cache = getRuntimeCache();
+    auto result = cache->getOrCreate(key, builder);
+
+    execPtr = result.first;
+
+    if (execPtr) {
+        if (auto execInt8 = std::dynamic_pointer_cast<DeconvExecutorInt8>(execPtr)) {
+            if (internalBlobMemory.empty() || internalBlobMemory.front()->GetPrimitive().get_desc() != execInt8->getWeightDesc()) {
+                prepareMemory({MKLDNNExtensionUtils::makeDescriptor(execInt8->getWeightDesc())});
```

Comment:
1. Suppose weight cache is enabled. From its sources we can see that it does not check the tensor descriptor. But as far as I understand, this call placed here exactly to reorder the weights into the appropriate format. But what if instead of reordering the data will be simply read from the cache? Is such a situation possible?
2. Suppose we finally removed the internal blobs, and how it would work in that circumstances? I mean we would try to deduce the appropriate weight format and place it into the port config, just like we do it in conventional convolution, and only if we can not create a primitive with such weights format in runtime we will fallback on implicit reordering. So what is the difference here? Why we do not reorder the internal blob in advance using the most probable weight format based on the dummy shape with implicit reordering if required?

Reply:
Moved on `createPrimitive` stage. Now weight layout deduced based on dummy input, output shapes for dynamic case.
2nd problem have fixed.

1st problem also have fixed, because now for dynamic shape we create primitive with the same input, output shapes in all execution graphs. So if we take weights from cache we will take it with expected layout.
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/mkldnn_rnn.cpp#L862
@nshchego looks like 1st situation also possible for `MKLDNNRNN` node.

Please take a look.